libraries/type-length-value/README.md

```diff
@@ -45,57 +44,63 @@ impl SplDiscriminate for MyOtherPodValue {
 // and enough room for a `MyPodValue` and a `MyOtherPodValue`
-let account_size = TlvState::get_base_len() + std::mem::size_of::<MyPodValue>() + \
-    TlvState::get_base_len() + std::mem::size_of::<MyOtherPodValue>();
+let account_size = TlvStateMut::get_base_len()
+    + std::mem::size_of::<MyPodValue>()
+    + TlvStateMut::get_base_len()
+    + std::mem::size_of::<MyOtherPodValue>()
+    + TlvStateMut::get_base_len()
+    + std::mem::size_of::<MyOtherPodValue>()
 
 // Buffer likely comes from a Solana `solana_program::account_info::AccountInfo`,
 // but this example just uses a vector.
 let mut buffer = vec![0; account_size];
 
-// Unpack the base buffer as a TLV structure
-let mut state = TlvStateMut::unpack(&mut buffer).unwrap();
-
-// Init and write default value
-// Note: you'll need to provide a boolean whether or not to allow repeating
-// values with the same TLV discriminator.
-// If set to false, this function will error when an existing entry is detected.
-let value = state.init_value::<MyPodValue>(false).unwrap();
-// Update it in-place
-value.data[0] = 1;
-
-// Init and write another default value
-// This time, we're going to allow repeating values.
-let other_value1 = state.init_value::<MyOtherPodValue>(true).unwrap();
-assert_eq!(other_value1.data, 10);
-// Update it in-place
-other_value1.data = 2;
-
-// Let's do it again, since we can now have repeating values!
-let other_value2 = state.init_value::<MyOtherPodValue>(true).unwrap();
-assert_eq!(other_value2.data, 10);
-// Update it in-place
-other_value2.data = 4;
-
-// Later on, to work with it again, since we did _not_ allow repeating entries,
-// we can just get the first value we encounter.
-let value = state.get_first_value_mut::<MyPodValue>().unwrap();
+{
+    // Unpack the base buffer as a TLV structure
+    let mut state = TlvStateMut::unpack(&mut buffer).unwrap();
+    
+    // Init and write default value
+    // Note: you'll need to provide a boolean whether or not to allow repeating
+    // values with the same TLV discriminator.
+    // If set to false, this function will error when an existing entry is detected.
+    let (value, _) = state.init_value::<MyPodValue>(false).unwrap();
+    // Update it in-place
+    value.data[0] = 1;
+    
+    // Init and write another default value
+    // This time, we're going to allow repeating values.
+    let (other_value1, _) = state.init_value::<MyOtherPodValue>(true).unwrap();
+    assert_eq!(other_value1.data, 10);
+    // Update it in-place
+    other_value1.data = 2;
+    
+    // Let's do it again, since we can now have repeating values!
+    let (other_value2, _) = state.init_value::<MyOtherPodValue>(true).unwrap();
+    assert_eq!(other_value2.data, 10);
+    // Update it in-place
+    other_value2.data = 4;
+    
+    // Later on, to work with it again, since we did _not_ allow repeating entries,
+    // we can just get the first value we encounter.
```

Comment:
Do you mind updating this comment to mention that specifically the `MyPodValue` type is the one we didn't allow repeats with? I think just rewording it is good.

Reply:
Done!